src/managers/GCommandLoader.js

```diff
@@ -84,17 +86,25 @@ class GCommandLoader {
      * @private
      */
     async __loadCommandCategoryFiles(categoryFolder) {
-        for await (let file of (await fs.readdirSync(`${this.cmdDir}/${categoryFolder}`))) {
-            const fileName = file.split('.').reverse()[1];
-            const fileType = file.split('.').reverse()[0];
+        for await (let fsDirent of (await fs.readdirSync(`${this.cmdDir}/${categoryFolder}`, { withFileTypes: true }))) {
+            let file = fsDirent.name;
+            // Note: fileName includes the extension (.)
+            const fileName = path.basename(file);
+            const fileType = path.extname(file);
```

Comment:
Same as the first two suggestions. Might be a good idea to abstract some of this logic into a separate function to reduce duplicating code too.

Reply:
Not looking into changing the structure of the module to any degree, would rather leave that up to others.

Reply:
That's understandable and completely fine.